src/routing_table/mod.rs

```diff
@@ -259,10 +259,32 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         }
     }
 
+    pub fn new_with_prefixes<U: IntoIterator<Item=Prefix<T>>>(our_name: T, min_group_size: usize, prefixes: U) -> Self {
+        let mut groups = HashMap::new();
+        let mut our_group_prefix = Prefix::new(0, our_name);
+        for prefix in prefixes {
+            if prefix.matches(&our_name) && prefix.bit_count() > our_group_prefix.bit_count() {
+                our_group_prefix = prefix;
+            }
+            let _ = groups.insert(prefix, HashSet::new());
```

Comment:
Since this is now not test-only, we should probably validate that the given prefixes make sense, i.e. that they satisfy the invariant, are disjoint, etc.